lib/server.js

```diff
@@ -23,9 +23,16 @@ module.exports = class HTTPServer extends TCPServer {
 
   setTimeout (ms = 0, ontimeout) {
     if (ontimeout) this.on('timeout', ontimeout)
-
     this._timeout = ms
-
     return this
   }
+
+  close () {
+    this._closeConnections()
+    super.close()
+  }
+
+  _closeConnections () {
+    for (const socket of this._connections) socket.end()
```

Comment:
Almost there! This should only close _idle_ connections, i.e. the sockets that aren't actively used for a request. Requests that are still in-progress should be allowed to finish and _then_ those connections should be closed as well.